text/0040-token-metadata.md

```diff
@@ -0,0 +1,163 @@
+- Feature Name: `token_metadata`
+- Start Date: May 16 2022
+- MCIP PR: [mobilecoinfoundation/mcips#40](https://github.com/mobilecoinfoundation/mcips/pull/40)
+- Tracking Issue: (none yet)
+
+# Summary
+[summary]: #summary
+
+Specify a source of truth for metadata about new tokens, and a way for clients to discover this,
+by querying the consensus nodes.
+
+# Motivation
+[motivation]: #motivation
+
+[MCIP #25](https://github.com/mobilecoinfoundation/mcips/pull/0025) and [MCIP #37](https://github.com/mobilecoinfoundation/mcips/pull/0037)
+have specified ways for new tokens to exist on chain. However, for user-facing applications to make effective use of this feature,
+they need more information about a token than just a token id -- they ultimately need to know how they should present balance information
+to the user.
+
+One way to do this is to hard-code all metadata into clients, which might work initially. After this however, this will mean that new tokens
+that are created on chain can't actually be used by users until a client update has been released and the users have updated.
+
+(In the [MCIP #31](https://github.com/mobilecoinfoundation/mcips/pull/0025) implementation PR, we did add a proto enum called `KnownTokenIds`
+which can be shipped with clients, as a stop-gap measure.)
+
+If the clients can dynamically query this information, then a client update is not required for clients to make use of new tokens which are
+created, which significantly reduces the amount of time it takes for a new token to be available to users.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Token metadata includes the following:
+
+* Short (exchange ticker) name of a token, e.g. MOB. (This is not expected to be translated.)
+* Long (english) name of a token, e.g. MobileCoin. (This is expected to be translated, but the client must maintain the translation list.)
+* Display conversion factor, e.g. 10^12. This conversion factor converts from the smallest representable units (`u64`) which appear in the blockchain,
```

Comment:
i think for bitcoin, the fundamental unit is satoshi?

i think in algorand standard assets, they have a "unit name" field, so that could be e.g. picoMob, satoshi, etc. for us?

Reply:
i am not sure how SI prefix stuff interacts with translations, so like, i think usually you could not translate `pico`, then translate `Mob`, then concatenate them

Reply:
https://www.metricmethods.com/Resources.php

[Chinese](https://www.metricmethods.com/SI_chinese.html)	[French](https://www.metricmethods.com/SI_french.html)	[German](https://www.metricmethods.com/SI_german.html)	[Greek](https://www.metricmethods.com/SI_greek.html)
[Italian](https://www.metricmethods.com/SI_italian.html)	[Japanese](https://www.metricmethods.com/SI_japanese.html)	[Russian](https://www.metricmethods.com/SI_russian.html)	[Spanish](https://www.metricmethods.com/SI_spanish.html)

Im ok with unit_name, but that could end up being less translatable than an SI prefix.

Reply:
thinking more about it, probably best to choose the most flexible option which would be a `unit_name` field